src/main/java/xades4j/xml/unmarshalling/FromXmlBaseTimeStampConverter.java

```diff
@@ -60,7 +61,7 @@ protected void convertTimeStamps(
             }
             else
             {
-                List params = CollectionUtils.filterByType(xmlCanonMethod.getContent(), Element.class);
+                List<Node> params = CollectionUtils.filterByType(xmlCanonMethod.getContent(), Node.class);

```

Comment:
This is not equivalent, even though the constructor below takes `List<Node>`. Please revert.

Reply:
we have an unchecked assignment than. Would an explicit cast be better:
```suggestion
                List<Element> params = CollectionUtils.filterByType(xmlCanonMethod.getContent(), Element.class);
                c14n = new GenericAlgorithm(xmlCanonMethod.getAlgorithm(), (Node) params);
```

Reply:
We can use List<Element> and change the constructor below (`GenericAlgorithm`) to take `List<? extends Node> params`, right?

Reply:
Perfect! I didn't see this!